src/watch_state.rs

```diff
@@ -42,33 +42,48 @@ impl WatchState {
         // https://docs.rs/watchexec/latest/watchexec/action/enum.Outcome.html
         let outcome = match (effect, *state) {
             (Effect::Exit, _) => {
-                Outcome::if_running(Outcome::both(Outcome::Stop, Outcome::Exit), Outcome::Exit)
+                Outcome::if_running(Outcome::both(stop_outcome(), Outcome::Exit), Outcome::Exit)
             }
             (Effect::ChildProcessFailed, _) => {
-                Outcome::if_running(Outcome::Stop, Outcome::DoNothing)
+                Outcome::if_running(stop_outcome(), Outcome::DoNothing)
             }
             (Effect::ChildProcessCompleted, State::Building) => {
                 *state = State::Running;
                 self.restart_outcome(PREVENT_CLEAR)
             }
-            (Effect::ChildProcessCompleted, State::Running) => Outcome::DoNothing,
+            (Effect::ChildProcessCompleted, State::Running) => {
+                *state = State::Building;
+                self.restart_outcome(PREVENT_CLEAR)
+            }
+            (Effect::ChildProcessCompleted, State::WaitingForSpinUpToExit) => {
+                *state = State::Building;
+                self.restart_outcome(PREVENT_CLEAR)
+            }
             (Effect::ManifestChange, State::Building) => self.restart_outcome(ALLOW_CLEAR),
+            (Effect::ManifestChange, State::WaitingForSpinUpToExit) => Outcome::DoNothing,
             (Effect::ManifestChange, State::Running) => {
                 if !self.skip_build {
-                    *state = State::Building;
+                    *state = State::WaitingForSpinUpToExit;
+                    return Outcome::if_running(stop_outcome(), Outcome::DoNothing);
                 }
                 self.restart_outcome(ALLOW_CLEAR)
             }
             (Effect::SourceChange, State::Building) => self.restart_outcome(ALLOW_CLEAR),
+            (Effect::SourceChange, State::WaitingForSpinUpToExit) => Outcome::DoNothing,
             (Effect::SourceChange, State::Running) => {
                 if !self.skip_build {
-                    *state = State::Building;
-                    return self.restart_outcome(ALLOW_CLEAR);
+                    *state = State::WaitingForSpinUpToExit;
+                    return Outcome::if_running(stop_outcome(), Outcome::DoNothing);
                 }
                 Outcome::DoNothing
             }
             (Effect::ArtifactChange, State::Building) => Outcome::DoNothing,
-            (Effect::ArtifactChange, State::Running) => self.restart_outcome(ALLOW_CLEAR),
+            (Effect::ArtifactChange, State::WaitingForSpinUpToExit) => Outcome::DoNothing,
+            (Effect::ArtifactChange, State::Running) => {
+                *state = State::WaitingForSpinUpToExit;
+                Outcome::if_running(stop_outcome(), Outcome::DoNothing)
```

Comment:
We produce this outcome in three locations. I wonder if it would read a little clearer if we created it with a function.